py3dtilers/Common/tileset_creation.py

```diff
@@ -103,11 +103,12 @@ def create_tile_content(objects, extension_name=None, with_texture=False, store_
 
     # Create a batch table and add the ID of each feature to it
     ids = [feature.get_id() for feature in objects]
+    data = [feature.get_data() for feature in objects]
     bt = BatchTable()
     bt.add_property_from_array("id", ids)
-    if store_alt_id:
-        alt_ids = [feature.get_alt_id() for feature in objects]
-        bt.add_property_from_array("alt_id", alt_ids)
+    # if there is application specific data associated with the object, add it to the batch table
+    if not all([d is None for d in data]):
+        bt.add_property_from_array("data", data)
```

Comment:
If I understand correctly, data is an array. 
Maybe we can change it to a dictionnary that contains additional data, along with a key. This would allow us to add (if needed) multiple semantic data in the batch table : 
"height" : [10, 9, 8], "gmlid":[545,546,684].

Also, there is a need to check if there is the same number of feature than cityobject

Reply:
The lengths should be the same since every object from object_to_tile.py has a `self.data` property even if it is set to None. It won't hurt to check it though.

Reply:
Oh I misunderstood what you meant by the structure of the batch table. Actually, the currently proposed data object is an array of _objects_, for example:
```
data = [ 
    { 'id' : 1, 'height' : 10 },
    {'id' : 2, 'height' : 10 },
    {'id' : 3, 'height' : 10 },
...]
```

Would this approach work with IFCTiler and the other tilers? I am also wondering if this is not an optimal way to structure the batch table.

Reply:
If you look [here](https://github.com/CesiumGS/3d-tiles/blob/main/specification/TileFormats/BatchTable/README.md#json-header), I understand that the structure of the batch table should be : 
 ```
data = [ 
    { 'id' : 1,  2, ... },
    { 'height' : 10, 11  , ... },
...]
```

It look quite similar to what you have done, and only need a few changes

Reply:
Ah I see. I'll change the structure so that it matches that structure.

Too bad, I liked the idea of storing each city object's batch table data as a single json object.

Reply:
Totally agree, it could help to parse the batch table more easily.
I think this choice was made so that attributes names are not repeated for each object to help reduce the size of the file. 

If you can find more practical use so that we need to change property to a single json object, maybe we can create an extension and propose it to 3DTiles in the futur.